addons/website/static/src/js/editor/widget_link.js

```diff
@@ -118,11 +118,14 @@ patch(LinkTools.prototype, 'website/static/src/js/editor/widget_link.js', {
         if (ev.currentTarget.closest('[name="link_anchor"]')) {
             const anchorValue = $(ev.currentTarget).data('value');
             const $urlInput = this.$el.find('[name="url"]');
+            const $selectMenu = this.$el.find('we-selection-items[name="link_anchor"]');
             let urlInputValue = $urlInput.val();
             if (urlInputValue.indexOf('#') > -1) {
                 urlInputValue = urlInputValue.substr(0, urlInputValue.indexOf('#'));
             }
-            $urlInput.val(urlInputValue + anchorValue);
+            const newUrlValue = urlInputValue + anchorValue;
+            $urlInput.val(newUrlValue);
+            $selectMenu.data('anchor-for', newUrlValue);
```

Comment:
Hello! When this is fw-ported to 17.0, be sure to give it a quick test.
You will probably need to include a call to `this.__onURLInput()` after these lines  in 17.0+ (see https://github.com/odoo/odoo/pull/140061).

Since the conversion of the Link widget to Owl in saas-16.4,  there's a `state.url` property, and since 17.0, the value of this property is used to update the link in the DOM. And `__onURLInput()` is the method that sets this property, based on the value of the URL input field.

Reply:
Noted!!! :+1: